### PR TITLE
(maint) Fix memory allocation for default facts C wrapper

### PR DESCRIPTION
**Pull request by MikaelSmith**

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I think this is right, not sure how we test the wrapper though.

File: lib/src/cwrapper.cc [hunk 1]
```diff
@@ -30,7 +30,7 @@ uint8_t get_default_facts(char **result) {
 
         auto json_facts = stream.str();
         auto l = json_facts.length()+1;
-        *result = new char [l];
+        *result = (char*)malloc(sizeof(char)*l);
```
Comment (branan) [inline lib/src/cwrapper.cc]: should use a C++-style case here (this is still C++ code).

**Comment by branan** [comment]
LGTM once the cast is fixed up

File: lib/src/cwrapper.cc (hunk 1 above)
Comment (MikaelSmith) [inline lib/src/cwrapper.cc]: Done.

File: lib/src/cwrapper.cc [hunk 2]
```diff
@@ -30,7 +30,7 @@ uint8_t get_default_facts(char **result) {
 
         auto json_facts = stream.str();
         auto l = json_facts.length()+1;
-        *result = new char [l];
+        *result = static_cast<char*>(malloc(sizeof(char)*l));
```
Comment (ericldelaney) [inline lib/src/cwrapper.cc]: Don't you need to check that the malloc didn't return NULL?

Comment (MikaelSmith) [inline lib/src/cwrapper.cc]: Urgh, yes.

**Pull request merged**